client/ayon_maya/plugins/publish/extract_layout.py

```diff
@@ -176,6 +180,33 @@ def process(self, instance):
         self.log.debug("Extracted instance '%s' to: %s",
                        instance.name, json_representation)
 
+    def process_container(self, containers):
+        """Allow to collect the asset containers through sub-assembly workflow if
+        there is a layout container
+
+        Args:
+            containers (list): Ayon asset containers
+
+        Returns:
+            dict: container dict with the container and its related asset name
+        """
+        all_containers_set = {}
+        for container in containers:
+            if cmds.getAttr(f"{container}.loader") == "LayoutLoader":
+                # Flatten container layout loader products to their individual loaded containers
+                for member_container in get_container_members(container):
+                    # Recursively process each member container
+                    self.process_container(member_container)
```

Comment:
This returns nothing - but the results should actually be 'updated' into the parent dict.

```suggestion
                    children_containers = self.process_container(member_container)
                    all_containers_set.update(children_containers)
```
I suppose something like this would be needed?

Reply:
Also, `process_containers` takes a `list[str]` - not a single container, so this call is wrong to begin with?